sway-core/src/parse_tree/expression/mod.rs

```diff
@@ -117,6 +117,10 @@ pub struct ReassignmentExpression {
 
 #[derive(Debug, Clone)]
 pub enum ExpressionKind {
+    /// A malformed expression.
+    ///
+    /// Used for parser recovery when we cannot form a more specific node.
+    Error,
```

Comment:
Is it possible to access the `Box<[Span]>` here? It looks like you have added them to `sway-ast` but not carried them forward into the `ParseProgram`. The `ParseProgram` is the lowest data we access in LSP so anything that isn't carried forward isn't accessible.

Reply:
Excellent thanks!